presto-spark-base/src/test/java/com/facebook/presto/spark/adaptive/execution/TestPrestoSparkAdaptiveJoinQueries.java

```diff
@@ -26,6 +35,29 @@ protected Session getSession()
     {
         return Session.builder(super.getSession())
                 .setSystemProperty(SPARK_ADAPTIVE_QUERY_EXECUTION_ENABLED, "true")
+                .setSystemProperty(ADAPTIVE_JOIN_SIDE_SWITCHING_ENABLED, "true")
                 .build();
     }
+
+    @DataProvider(name = "optimize_hash_generation")
+    public Object[][] optimizeHashGeneration()
+    {
+        return new Object[][] {{"true"}, {"false"}};
+    }
+
+    @Test(dataProvider = "optimize_hash_generation")
+    public void testQuerySucceedsWithAQE(String optimizeHashGeneration)
+    {
+        // we don't add a memory limit and test that the query succeeds with aqe and fails without
```

Comment:
Will a test like this work:

SELECT * FROM nation n JOIN (SELECT * FROM orders cross join unnest sequence(1, 50)) o ON n.nationkey = o.orderkey

Reply:
no, wouldn't help.